### DISCUSSION TITLE
404 whatever I do

### DISCUSSION THREAD

**Discussion by salmipa** [quarkiverse/quarkus-cxf#416] [General]
Hi,

I trying to make quarkus-cxf work, but whatever I do it never finds my endpoint. Just 404 all the time. 

this is in my pom.xml

    <dependency>
      <groupId>io.quarkiverse.cxf</groupId>
      <artifactId>quarkus-cxf</artifactId>
      <version>0.14.0</version>
    </dependency>

and this in application.properties

quarkus.http.port=9123
quarkus.cxf.path=/cxf
quarkus.cxf.endpoint."/fruit".implementor=blahblah.PolicySearchService

HTTP GET for 

http://localhost:9123/cxf/fruit?wsdl 

says Resource not found. Similarly POST with actual SOAP message says the same.

What's wrong?

**Comment by shumonsharif** [comment]
Hi @salmipa Are you able to put together and share a small reproducer?

If not, maybe you can try enable debug logging to have some additional insight.

```properties
# To enable additional logging for CXF and quarkus-cxf
quarkus.log.category."org.apache.cxf".level=DEBUG
quarkus.log.category."io.quarkiverse.cxf".level=DEBUG
```

**Reply by salmipa** [reply]
Hi,

And thanks for your reply. Nothing was printed out with the log settings until I changed my implementation so that I had SEI and implementation separated (used to be just implementation class) and all of the sudden it started to work. Is this somehow expected behavior?

Br, Paci

**Reply by shumonsharif** [reply]
Hi Paci, I believe you've uncovered an important issue; the extension currently expects an interface to be defined.  Looking deeper into the JAX-WS specifications, an interface is not _required_, just an implementation class should also work.  I will look into having this addressed in the next release.  Thank you!

**Reply by shumonsharif** [reply]
I have created https://github.com/quarkiverse/quarkus-cxf/issues/417 to track this.

**Reply by shumonsharif** [reply]
This issue is fixed in the latest release v1.0.0.
